library/core/src/fmt/mod.rs

```diff
@@ -190,7 +190,10 @@ pub trait Write {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     fn write_fmt(mut self: &mut Self, args: Arguments<'_>) -> Result {
-        write(&mut self, args)
+        match args.as_str() {
+            Some(s) => self.write_str(s),
+            None => write(&mut self, args),
```

Comment:
This special casing should probably be done inside `write` to cover more cases.

Reply:
Actually, it may already be done there.

Reply:
It seems that moving this logic to `write` doesn't help inline calls.